### ISSUE TITLE
building the runtime fails

### ISSUE THREAD

**Issue by dvc94ch** [AppImage/AppImageKit#1172]
```
cmake -DCMAKE_BUILD_TYPE=Release .
make runtime
```
fails with
```
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-ll.o):(.bss+0x0): multiple definition of `libhandle'; runtime.4.o:(.bss.libhandle+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-ll.o):(.bss+0x8): multiple definition of `have_libloaded'; runtime.4.o:(.bss.have_libloaded+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-ll.o):(.bss+0x10): multiple definition of `load_library_errmsg'; runtime.4.o:(.bss.load_library_errmsg+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-ll_inode.o):(.bss+0x0): multiple definition of `libhandle'; runtime.4.o:(.bss.libhandle+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-ll_inode.o):(.bss+0x8): multiple definition of `have_libloaded'; runtime.4.o:(.bss.have_libloaded+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-ll_inode.o):(.bss+0x10): multiple definition of `load_library_errmsg'; runtime.4.o:(.bss.load_library_errmsg+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-nonstd-daemon.o):(.bss+0x0): multiple definition of `libhandle'; runtime.4.o:(.bss.libhandle+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-nonstd-daemon.o):(.bss+0x8): multiple definition of `have_libloaded'; runtime.4.o:(.bss.have_libloaded+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libsquashfuse_ll_la-nonstd-daemon.o):(.bss+0x10): multiple definition of `load_library_errmsg'; runtime.4.o:(.bss.load_library_errmsg+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libfuseprivate_la-fuseprivate.o):(.bss+0x0): multiple definition of `libhandle'; runtime.4.o:(.bss.libhandle+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libfuseprivate_la-fuseprivate.o):(.bss+0x8): multiple definition of `have_libloaded'; runtime.4.o:(.bss.have_libloaded+0x0): first defined here
/usr/bin/ld: ../lib/libappimage/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse_ll.a(libfuseprivate_la-fuseprivate.o):(.bss+0x10): multiple definition of `load_library_errmsg'; runtime.4.o:(.bss.load_library_errmsg+0x0): first defined here
collect2: error: ld returned 1 exit status
```

**Comment by dvc94ch** [comment]
it's arch linux with a recent clang or gcc, don't know which one your build defaults to. extracted the runtime from an existing appimage using `./appimagetool-x86_64.AppImage --appimage-offset` and `dd if=appimagetool-x86_64.AppImage of=appimage-runtime iflag=count_bytes count=189632`, so that works for me.

**Comment by TheAssassin** [comment]
First of all, Arch, like any rolling distro (or actually anything beyond CentOS 7) is the poorest choice for a build env I could think of. It demonstrates that you don't know a lot about AppImages. https://docs.appimage.org/introduction/concepts.html

Second, you could've found the default build environment easily within the `ci` directory in this repository. It's all in there,

Lastly, the runtime is available on the release page. Be aware appimagetool does a lot of things, e.g., including update information. You do not benefit from this. In fact, your AppImage may carry broken information now. You should just have used the appimagetool AppImage from the release page, as it is documented virtually everywhere.

Bleeding edge compilers are not supported. Closing therefore.

**Comment by TheAssassin** [comment]
P.S.: Your initial report contained little to no useful information about your surrounding environment. No distro, no compiler, ..., nothing. I'd strongly recommend you to read https://www.chiark.greenend.org.uk/~sgtatham/bugs.html. Developers don't want to play information ping-pong in every single bug report. Please try to be more helpful and provide such information out of the box.

**Comment by dvc94ch** [comment]
ah indeed, the runtime is available as a release, thanks. however I find your tone condescending, and the runtime is a static lib, so completely irrelevant what system it was built on.

**Comment by TheAssassin** [comment]
The runtime is *not* a static binary (ignore what `ldd` says, it is offended by the magic bytes, AFAIR, use `lddtree` or another tool; `ldd` works on the intermediate build artifacts before they are patched, though). This assumption is plainly wrong. There is a reason why I just told you that what you tried to do is a bad idea. 

> I find your tone condescending

Well, that was intentional. You seem to be an active developer on GitHub, and I would expect someone who receives bug reports themselves to know what a poor bug report is. As someone who reads those kind of reports every day in time I donate to a variety of projects, I truly have better things to do than repeat myself on this topic over and over and over again. Our docs contain all the conceptual information and also guides on packaging, and our GitHub actions stuff is not hard to find at all.

I was a bit annoyed by your initial report due to the lack of information, and the report was quite offending, I just explained why. A newbie would not receive such a reply, because they might not know better. But in this case, I saw no reason to wrap my reply up in cotton wool. I've gotten much worse replies on much better reports before. My previous reply contains all the information you were likely looking for nevertheless.
